rails/app/controllers/authentications_controller.rb

```diff
@@ -87,12 +87,14 @@ def generic_oauth
         e,
         env: request.env,
         data: {
-          extra_first_name: extra_first_name, 
+          extra_first_name: extra_first_name,
           info_first_name: info_first_name,
           extra_last_name: extra_last_name,
           info_last_name: info_last_name,
           first_name_64: Base64.encode64(extra_first_name  || info_first_name || ''),
-          last_name_64: Base64.encode64(extra_last_name || info_last_name || '')
+          last_name_64: Base64.encode64(extra_last_name || info_last_name || ''),
+          provider: auth&.provider,
+          uid: auth&.uid
         }
```

Comment:
Looks good, bit at this point wouldn't it be better just to set data to `auth.to_json()` (after clearing any sensitive data from the object)?